src/sentry/monitors/tasks.py

```diff
@@ -92,21 +88,13 @@ def check_monitors(current_datetime=None):
         except Exception:
             logger.exception("Exception in check_monitors - mark missed")
 
-    qs = MonitorCheckIn.objects.filter(status=CheckInStatus.IN_PROGRESS).select_related(
-        "monitor", "monitor_environment"
-    )[:CHECKINS_LIMIT]
+    qs = MonitorCheckIn.objects.filter(
+        status=CheckInStatus.IN_PROGRESS, timeout_at__lte=current_datetime
```

Comment:
would it be worth changing the query to
`Q(timeout_at__lte=current_datetime) | Q(timeout_at=None)`
as a safety precaution? we don't have any null values currently but would be a good protection in case there's a regression so we don't ignore the null timeouts. could also put it in a separate query

Reply:
Hmmm. Maybe we can add a metric that we put an alert on?

Reply:
I'll run a query and raise an exception at the bottom as a safety check